archivessnake/create_daos.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+
+"""
+Using DIP information from Archivematica, adds digital object to file-level children of a series in ArchivesSpace.
+
+Expects an external pickle file that contains a list of dictionaries with the following information:
+    current_full_path: full path to DIP (from Archivematica)
+    related_packages: AIP URI (from Archivematica)
+    uuid: DIP UUID (from Archivematica)
+    box: parsed from current_full_path
+    folder: parsed from current_full_path
+"""
+
+import argparse
+import pickle
+from configparser import ConfigParser
+from os.path import join
+
+import requests
+from asnake.aspace import ASpace
+
+# set up archivesspace client
+config = ConfigParser()
+config.read("local_settings.cfg")
+
+parser = argparse.ArgumentParser(
+    description="Using DIP information from Archivematica, adds DAOs to file-level children of a series in ArchivesSpace")
+parser.add_argument(
+    'resource', help='ArchivesSpace ID of resource. Example: 123')
+parser.add_argument('series', help='ArchivesSpace ID of series. Example: 4321')
+parser.add_argument(
+    'dip_info_file',
+    help='Path to pickle file containing array of dictionaries with DIP information.')
+args = parser.parse_args()
+
+
+def main(resource, series, dip_file):
+    """Main function, which is run when this script is executed"""
+    aspace = ASpace(
+        baseurl=config.get(
+            "ArchivesSpace", "baseURL"), username=config.get(
+            "ArchivesSpace", "username"), password=config.get(
+                "ArchivesSpace", "password"))
+    dao_url = join(config.get("ArchivesSpace", "baseURL"),
+                   "repositories/2/digital_objects")
+    dip_data = load_dip_file(dip_file)
+    for uri in get_uris(aspace, resource, series):
+        try:
+            dip = match_dip_to_aip(aspace, uri, dip_data)
+            dip_uuid, title, aip_uuid = get_dip_info(dip)
+            dao_uri = post_dao(dao_url, aspace, dip_uuid, title, aip_uuid)
+            update_component(aspace, uri, dao_uri)
+        except Exception as e:
+            print(e)
```

Comment:
Do I need to, since this is a `for` loop? It did not cause issues when I ran it.

Reply:
Yeah, that's true, you're already in a for loop. Are there other things you'd need to do to clean up if something went wrong? The only thing that occurs to me is if your `update_component` fails, you're going to have an orphaned digital object hanging out in AS.

Reply:
I've updated the error message when a DAO cannot be linked to log exact information; there are cases where we may want to "clean up" this type of issue by linking the DAOs in a separate process instead of deleting them.